source_sdk/entity/entity.c

```diff
@@ -32,10 +32,10 @@ __int32_t get_ent_class(void *entity)
 
 struct vec3_t get_ent_origin(void *entity)
 {
-    // x + 0x340, y + 0x344, z + 0x348
-    return *(struct vec3_t *)((__uint64_t)(entity) + 0x340);
-}
-
+    // x + 0x328, y + 0x332, z + 0x346
+    return *(struct vec3_t *)((__uint64_t)(entity) + 0x328); //PR comment: This is the correct offset. The one which was used prior is used for world origin.
```

Comment:
Nice find. This works great for hitscan weapons and better for projectile aimbot too. At the moment the projectile aimbot misses at large distances, I think eventually the goal is to use `EstimateAbsVelocity` for the projectile prediction.